pkg/cfg/cfg.go

```diff
@@ -353,6 +355,15 @@ func basicTest() error {
 	if Cfg.Cookie.MaxAge > Cfg.JWT.MaxAge {
 		return fmt.Errorf("configuration error: Cookie maxAge (%d) cannot be larger than the JWT maxAge (%d)", Cfg.Cookie.MaxAge, Cfg.JWT.MaxAge)
 	}
+	// if using regexWhiteList, make sure that all the regex statements will compile
+        if len(Cfg.RegexWhiteList) != 0 {
+                for _, wl := range Cfg.RegexWhiteList {
+                        _, reWhiteListErr := regexp.Compile(wl)
```

Comment:
How about storing the compiled regex for later usage to avoid the `/validate` compile for each regex.

Might be worthwhile to sort the list by length, longest first.  Or perhaps clarify in the order in the docs (config_example.yml).

Reply:
> Might be worthwhile to sort the list by length, longest first. Or perhaps clarify the order in the docs (config_example.yml).

Do you mean sort the list of regex statements and then compile them in sorted order? We could do that, but my preference is that the order of execution is left to the user. Then again, since there is no "fail" condition, only that there is no match in the list at all. Whatever results in the least processing per auth should probably be preferred.

The question becomes, what would be statistically more likely to match with less cpu overhead?
- Sort from longest regex to shortest (may or may not be more complex)
- Sort from shortest to longest
- Do not sort and the order in which the user enters them is the order of execution.

The "best" answer is probably benchmark the regex against a test set and determine the order that, but that seems overly complex for what we are trying to accomplish.